pkg/app/piped/livestatestore/cloudrun/store.go

```diff
@@ -30,10 +32,74 @@ type store struct {
 	client provider.Client
 }
 
+type app struct {
+	service provider.ServiceManifest
+	// The states of service and all its active revsions which may handle the traffic.
+	states  []*model.CloudRunResourceState
+	version model.ApplicationLiveStateVersion
+}
+
 func (s *store) run(ctx context.Context) error {
+	svcs, err := s.fetchManagedServices(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to fetch managed services: %w", err)
+	}
+
+	revs := make(map[string][]*provider.Revision, len(svcs))
+	for i := range svcs {
+		id, ok := svcs[i].UID()
+		if !ok {
+			continue
+		}
+		names := svcs[i].ActiveRevisionNames()
+		rs, err := s.fetchActiveRevisions(ctx, names)
+		if err != nil {
+			return fmt.Errorf("failed to fetch active revisions: %w", err)
+		}
+		if len(rs) == 0 {
+			continue
+		}
+		revs[id] = rs
+	}
+
+	// Update apps to the latest.
+	apps := s.buildAppMap(svcs, revs)
+	s.apps.Store(apps)
+
+	return nil
+}
+
+func (s *store) buildAppMap(svcs []*provider.Service, revs map[string][]*provider.Revision) map[string]app {
+	apps := make(map[string]app, len(svcs))
+	for i := range svcs {
+		sm, err := svcs[i].ServiceManifest()
+		if err != nil {
+			s.logger.Error("failed to load cloudrun service into service manifest", zap.Error(err))
+			continue
+		}
+
+		appID, ok := sm.AppID()
+		if !ok {
+			continue
+		}
+
+		id, _ := svcs[i].UID()
+		now := time.Now()
+		apps[appID] = app{
+			service: sm,
+			states:  provider.MakeResourceStates(svcs[i], revs[id], now),
+			version: model.ApplicationLiveStateVersion{
+				Timestamp: now.Unix(),
+			},
+		}
+	}
```

Comment:
To make sure that all resources have the same version.

```suggestion
        now := time.Now()
        version: model.ApplicationLiveStateVersion{
                Timestamp: now.Unix(),
        }
	for _, svc := range svcs {
		sm, err := svc.ServiceManifest()
		if err != nil {
			s.logger.Error("failed to load cloudrun service into service manifest", zap.Error(err))
			continue
		}

		appID, ok := sm.AppID()
		if !ok {
			continue
		}

		id, _ := svc.UID()
		apps[appID] = app{
			service: sm,
			states:  provider.MakeResourceStates(svc, revs[id], now),
			version: version,
		}
	}
```